### PR TITLE
CoinTracker : React Capstone Project

### PR DESCRIPTION
**Pull request by ohyajk**
CoinTracker shows data on the top 100 cryptocurrencies around the world. It is built using the following features :-

- Create React App
- Redux Toolkit (Thunk & Logger)
- TailwindCSS
- React Router
- React Testing Library
- Jest
- EsLint & Style Linters

Liver Demo Link  : https://cointracker-q7ta.onrender.com
Loom video Link : https://www.loom.com/share/5a8587dc420a4eb492c32e482ea7e051

Finally, I want to extend my regards to Coinstats for providing API for this project.
Additional credits are included in the Readme file.

KungFuPandaReviewGIF

### CODE REVIEW

File: src/__test__/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+test('should render snap', () => {
+  render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Home />
+      </BrowserRouter>
+    </Provider>,
+  );
+  expect(screen.getByTestId('home-1')).toHaveTextContent('Snap');
+});
```
Comment (jerryowusu) [inline src/__test__/Home.test.js:28]: Nice work for testing almost all the components. However, as part of the project requirements, you should also add test for some pure functions in redux code. Your app should also have integration test using React Testing Library. :pray:

File: src/Components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        {!allCoins && isLoading === false ? (
+          <div className="text-center text-2xl flex items-center justify-center h-[40vh] ">
+            <span>
+              <span className="changer">Oh Snap !</span>
+              {' '}
+              We have a problem...
+            </span>
+          </div>
+        ) : ''}
+      </div>
```
Comment (jerryowusu) [inline src/Components/Home.jsx]: Nice and beuatiful design so far, however, as stated in the project requirements, you are suppose to follow the design guidelines including: 

- [x] Colors (select one main color for your website).
- [x] Typography: font face, size and weight.
- [x] Layout: composition and space between elements.

There are some differences between yours and that of the actual design which you will need to fix.

1.  The home page is actually suppose to be in two rows with a light and dark color alternating between the cards on each column.  

2.  Kindly check the size and position of the country images, the country name and numbers below, as well as the small arrows on the top right corner of each country card. 


### Your design 

Screenshot from 2023-02-07 18-48-55


### The actual design

Screenshot from 2023-02-07 18-49-55

File: src/Components/Details.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      </h4>
+      <h4>
+        Price Change Daily :
+        {details.priceChange1d < 0 ? <span className="text-red-400">{details.priceChange1d}</span> : <span className="text-green-400">{details.priceChange1d}</span>}
+      </h4>
+      <h4>
+        Price Change Hourly :
+        {details.priceChange1h < 0 ? <span className="text-red-400">{details.priceChange1h}</span> : <span className="text-green-400">{details.priceChange1h}</span>}
+      </h4>
+    </div>
```
Comment (jerryowusu) [inline src/Components/Details.jsx]: Nice and beuatiful design so far, however, as stated in the project requirements, you are suppose to follow the design guidelines including: 

- [x] Colors (select one main color for your website).
- [x] Typography: font face, size and weight.
- [x] Layout: composition and space between elements.

There are some differences between yours and that of the actual design which you will need to fix.

1. The background of each odd card is darker than its preceeding (even) card forming a checkered layout.
2. Also, kindly note the margin, paddings, and distance between the cities name and the views
3. Your design is also missing the small arrow to the right of each card. 

### Your design 

Screenshot from 2023-02-07 18-49-08


### The actual design 

Screenshot from 2023-02-07 18-50-02

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @ohyajk , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so close


## Highlights
Great effort so far :+1: 
No linter errors :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/Details.jsx (hunk 3 above)
Comment (ohyajk) [reply src/Components/Details.jsx]: Thank you for your review @jerryowusu I have done my best to implement the requested style changes.

Screenshot 2023-02-08 020622

File: src/Components/Home.jsx (hunk 2 above)
Comment (ohyajk) [reply src/Components/Home.jsx]: Thank you for your review @jerryowusu I have done my best to implement the requested style changes.
Screenshot 2023-02-08 021052

File: src/__test__/Home.test.js (hunk 1 above)
Comment (ohyajk) [reply src/__test__/Home.test.js:28]: Thank you @jerryowusu for your suggestion. I have added reducer tests to my project. 👍

**Review by jerryowusu — APPROVED** [review]
Hi @ohyajk , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

clapping-leonardo-dicaprio


## Highlights
Great effort in making the requested changes :+1: 
No linter errors :+1: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**